sdk/src/main/java/com/uid2/EUIDManager.kt

```diff
@@ -0,0 +1,107 @@
+package com.uid2
+
+import android.content.Context
+import com.uid2.UID2Manager.Companion
+import com.uid2.UID2Manager.Companion.APPLICATION_ID_DEFAULT
+import com.uid2.UID2Manager.Companion.UID2_API_URL_DEFAULT
+import com.uid2.network.DefaultNetworkSession
+import com.uid2.network.NetworkSession
+import com.uid2.storage.FileStorageManager
+import com.uid2.storage.FileStorageManager.Store.EUID
+import com.uid2.storage.StorageManager
+import com.uid2.utils.InputUtils
+import com.uid2.utils.Logger
+import com.uid2.utils.TimeUtils
+import kotlinx.coroutines.Dispatchers
+
+public class EUIDManager {
+
+    public companion object {
+        public data class Environment private constructor(
+            val serverUrl: String
+        ) {
+            public companion object {
+                // AWS EU West 2 (London)
+                public val london: Environment = Environment("https://prod.euid.eu/v2")
+
+                // Equivalent to `london`
+                public val production: Environment = london
+                public fun custom(serverUrl: String): Environment {
+                    return Environment(serverUrl)
+                }
+            }
+        }
+
+        private var serverUrl: String = UID2_API_URL_DEFAULT
+        private var applicationId: String = APPLICATION_ID_DEFAULT
+        private var networkSession: NetworkSession = DefaultNetworkSession()
+        private var storageManager: StorageManager? = null
+        private var isLoggingEnabled: Boolean = false
+
+        private var instance: UID2Manager? = null
+
+        /**
+         * Initializes the class with the given [Context], along with a [NetworkSession] that will be responsible
+         * for making any required network calls.
+         *
+         * @param context The context to initialise from. This will be used to obtain the package's metadata to extract
+         * the API URL.
+         * @param networkSession A custom [NetworkSession] which can be used for making any required network calls.
+         * The default implementation supported by the SDK can be found as [DefaultNetworkSession].
+         */
+        @JvmStatic
+        @JvmOverloads
+        @Throws(InitializationException::class)
+        public fun init(
```

Comment:
I feel a little nervous about this duplicated initialisation code, but don't (yet) know what else to suggest. It's due to the fact we are using singletons, rather than just provide a simple factory and for consumers to self-manage the instance. Maybe something we can revisit in the future.